bsp4j/src/main/java/ch/epfl/scala/bsp4j/Bsp4j.java

```diff
@@ -0,0 +1,5 @@
+package ch.epfl.scala.bsp4j;
+
+public class Bsp4j {
+    public static final String PROTOCOL_VERSION = "2.1.0";
```

Comment:
This might be problematic, as this can be inlined by the Java compiler resulting an a potential wrong value at runtime. A static method can avoid this.

Reply:
Okay, I've just read Puzzle 93 from Bloch's book :) fixing it now

Reply:
If I compile my library against version A, the compiler is allowed to inline the value "2.1.0" in the compiled binary code of the library. When I now use a newer version which would return "2.1.1", but do not recompile my library, the code which uses the value e.g. to construct a string, might still use the stale inlined "2.1.0" value. This is a general problem with constants in Java, hence constants which are suspected to change in a subsequent release (arbitrary but fixed), should not use a constant field but a constant method.

```suggestion
    public static final String protocolVersion() { return "2.1.0"; }
```

Reply:
TIL 😄